nixos/modules/services/misc/prowlarr.nix

```diff
@@ -27,13 +39,22 @@ in
 
       serviceConfig = {
         Type = "simple";
-        DynamicUser = true;
+        User = cfg.user;
+        Group = cfg.group;
         StateDirectory = "prowlarr";
         ExecStart = "${pkgs.prowlarr}/bin/Prowlarr -nobrowser -data=/var/lib/prowlarr";
```

Comment:
The dynamic user gets its state directory here. If it really needs a HOME directory, then that is just another environment variable.